src/sentry/snuba/discover.py

```diff
@@ -906,13 +922,22 @@ def top_events_timeseries(
     translated_groupby.sort()
 
     results = {}
-    for row in result["data"]:
-        result_key = create_result_key(row, translated_groupby, issues)
-        results.setdefault(result_key, {"data": []})["data"].append(row)
     # Using the top events add the order to the results
     for index, item in enumerate(top_events["data"]):
         result_key = create_result_key(item, translated_groupby, issues)
-        results[result_key]["order"] = index
+        results[result_key] = {
+            "order": index,
+            "data": [],
+        }
+    for row in result["data"]:
+        result_key = create_result_key(row, translated_groupby, issues)
+        if result_key in results:
+            results[result_key]["data"].append(row)
+        else:
+            logger.warning(
+                "Timeseries top events key mismatch",
```

Comment:
Most of our log messages are `.` and `-` separated. This could be `discover.top-events.timeseries.key-mismatch`

Reply:
👀 dang now you say that I notice it, thanks 🙏